library/rsa.c

```diff
@@ -1952,61 +1927,8 @@ int mbedtls_rsa_rsassa_pss_sign( mbedtls_rsa_context *ctx,
     else
         slen = olen - hlen - 2;
```

Comment:
Would it make sense to put this calculation in the new `mbedtls_rsa_rsassa_pss_sign_ext` function instead, but only use the calculation if a 0 salt length is passed? Then this function could blindly call the new function with a 0 salt len without having to do additional checking of the input parameters and parsing of the context.

Reply:
That's a good idea, thanks! I was struggling a bit with that in fact. Was not sure what the policy on validating input was, so I took the extra safe route. I'll take care of it over this weekend.

Reply:
Now that I think about it, are there use cases where one could request the salt to be 0 length? I don't remember from the top of my head what is covered in the CAVP test suite, and I'll check. But if this is a use case that should be supported then maybe using dedicated negative sentinel value (like -1 or something) instead of 0 could make sense. What do you think?

Reply:
Yes, the salt can be 0 length. In my limited experience, it's almost always the hash size, and if not it's 0. I seem to recall APIs that take a boolean: either no salt, or salt whose size is the hash size. PKCS#11 says “typical values are the length of the message hash and zero”.

Reply:
Maybe the new function can take both a boolean and a saltlen? False means ignore saltlen and calculate it, true means use saltlen. Then the old function just call the new function with (false, 0). Other users who want to use a 0 saltlen can call it with (true, 0).

Reply:
I'd prefer to use the interface that already exists for verification: pass the salt length as an `int` (it isn't going to overflow). Note that `MBEDTLS_RSA_SALT_LEN_ANY = -1` for verification means “accept any salt”. For signature, I think this should be the default value, and should be the current behavior.